lib/Onyx.js

```diff
@@ -189,15 +189,22 @@ function keysChanged(collectionKey, collection) {
                 });
             }
         } else if (isSubscribedToCollectionMemberKey) {
-            subscriber.withOnyxInstance.setState((prevState) => {
-                let newData = collection[subscriber.key];
-                if (_.isObject(newData)) {
-                    newData = lodashMerge(newData, prevState[subscriber.statePropertyName]);
-                }
-                return {
-                    [subscriber.statePropertyName]: newData,
-                };
-            });
+            const dataFromCollection = collection[subscriber.key];
+
+            // If `dataFromCollection` happens to not exist, then return early so that there are no unnecessary
+            // re-renderings of the component
+            if (_.isUndefined(dataFromCollection)) {
+                return;
+            }
+
+            subscriber.withOnyxInstance.setState(prevState => ({
+                [subscriber.statePropertyName]: _.isObject(dataFromCollection)
+                    ? {
+                        ...prevState[subscriber.statePropertyName],
```

Comment:
This seems potentially dangerous as it assumes we will always have at least an empty object for `prevState[subscriber.statePropertyName]` if we are dealing with object shaped data. If we try to spread something like `undefined` or `null` then we get 💥.

Maybe some combination of `lodash.merge()` could still be helpful here to help guard against that?

Reply:
Maybe there's no problem with that actually. I could just have made that up in my head.

Reply:
I think spreading `undefined` in objects works OK but arrays doesn't.

Reply:
I guess the small difference between this and using `lodashMerge()` is that the spread is not going to recursively merge the data. So depending on how complicated that data is we could get some weird results. But we can address that if it becomes a problem.

Reply:
Cool, thanks Marc. I think it'll be OK for now. Seeing as how everyone has approved this, I'm going to go ahead and merge it.